### PR TITLE
Maxwell dcconduction

### PR DESCRIPTION
**Pull request by IreneWoyna**
A new Maxwell 2D DC conduction example. I need a location to store the DXF file

### CODE REVIEW

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Plot electric field and current density on the conductor surface
+
+conductor_surface = m2d.modeler.get_object_faces(conductor.name)
+m2d.post.create_fieldplot_surface(conductor_surface, "Mag_E", plot_name="Electric Field")
+m2d.post.create_fieldplot_surface(conductor_surface, "Mag_J", plot_name="Current Density")
+
+##################################################################################
+# Release desktop
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:136]:
```suggestion
# Release desktop
# ~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    report_category="DCConduction",
+    plot_type="Data Table",
+    variations=variations,
+    plotname="Resistance vs. Material",
+)
+
+
+##################################################################################
+# Field overlay
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:127]:
```suggestion
# Field overlay
# ~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    "MaterialIndex", 0, no_materials-1, 1, "LinearStep",
+    parametricname="MaterialSweep")
+param_sweep["SaveFields"] = True
+param_sweep["CopyMesh"] = True
+param_sweep["SolveWithCopiedMeshOnly"] = True
+param_sweep.analyze()
+
+##################################################################################
+# Create resistance report
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:111]:
```suggestion
# Create resistance report
# ~~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    report_type="DCConduction",
+    expressions="1/Matrix1.G(1V,1V)/MaterialThickness",
+    isconvergence=True,
+    conv_criteria=1,
+    use_cache_for_freq=False)
+setup1.analyze()
+
+##################################################################################
+# Create parametric sweep
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:97]:
```suggestion
# Create parametric sweep
# ~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+##################################################################################
+# Assign mesh operation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 3mm on the conductor
+
+m2d.mesh.assign_length_mesh(["ANSYS_LOGO_2D_3"], meshop_name="conductor", maxlength=3, maxel=None)
+
+##################################################################################
+# Create simulation setup and enable expression cache
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:82]:
```suggestion
# Create simulation setup and enable expression cache
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+##################################################################################
+# Setup conductance calculation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 1V is the source, 0V ground
+
+m2d.assign_matrix(sources=['1V'], group_sources=['0V'], matrix_name="Matrix1")
+
+##################################################################################
+# Assign mesh operation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:75]:
```suggestion
# Assign mesh operation
# ~~~~~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+# Assign voltages
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 1V and 0V
+
+m2d.assign_voltage(["ANSYS_LOGO_2D_1"], amplitude=1, name="1V")
+m2d.assign_voltage(["ANSYS_LOGO_2D_2"], amplitude=0, name="0V")
+
+##################################################################################
+# Setup conductance calculation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:68]:
```suggestion
# Setup conductance calculation
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+# gets the material defined by the 0th entry of the material array
+
+m2d.assign_material(["ANSYS_LOGO_2D_1", "ANSYS_LOGO_2D_2"], "pec")
+conductor = m2d.modeler.get_object_from_name("ANSYS_LOGO_2D_3")
+conductor.material_name = "ConductorMaterial[MaterialIndex]"
+
+
+##################################################################################
+# Assign voltages
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:60]:
```suggestion
# Assign voltages
# ~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+# Import geometry as a DXF file
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+DXFPath = "D:/Ansys_logo_2D.dxf" # store this somewhere?
+dxf_layers = m2d.get_dxf_layers(DXFPath)
+m2d.import_dxf(DXFPath, dxf_layers, scale=1E-05)
+
+##################################################################################
+# Define variables
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:36]:
```suggestion
# Define variables
# ~~~~~~~~~~~~~~~~~
```

File: examples/03-Maxwell/Maxwell2D_DCConduction.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    specified_version="2023.1",
+    new_desktop_session=True,
+    close_on_exit=True,
+    solution_type="DCConduction",
+    designname="Ansys_resistor"
+)
+
+##################################################################################
+# Import geometry as a DXF file
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/03-Maxwell/Maxwell2D_DCConduction.py:28]:
```suggestion
# Import geometry as a DXF file
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by maxcapodi78 — APPROVED** [review]
Thanks. LGTM

**Pull request merged**